token-metadata/js/test/instruction.test.ts

```diff
@@ -0,0 +1,147 @@
+import { PublicKey, TransactionInstruction } from '@solana/web3.js';
+import { expect } from 'chai';
+
+import {
+    createEmitInstruction,
+    createInitializeInstruction,
+    createRemoveKeyInstruction,
+    createUpdateAuthorityInstruction,
+    createUpdateFieldInstruction,
+} from '../src';
+
+// Buffer outputs are verified against rust implimentation
+// To print packed values in rust file:
+// Add `use std::fmt::Write;` to the top of instructions.rs and
+// use `println!("{:?}", &packed);` inside check_pack_unpack to print buffer
+// Run with `cargo test -- --nocapture`
+
+describe('Token Metadata Instructions', () => {
+    const programId = new PublicKey('22222222222222222222222222222222222222222222');
+    const metadata = new PublicKey('33333333333333333333333333333333333333333333');
+    const updateAuthority = new PublicKey('44444444444444444444444444444444444444444444');
+    const mint = new PublicKey('55555555555555555555555555555555555555555555');
+    const mintAuthority = new PublicKey('66666666666666666666666666666666666666666666');
+
+    it('Can create Initialize Instruction', () => {
+        const instruction = createInitializeInstruction({
+            programId,
+            metadata,
+            updateAuthority,
+            mint,
+            mintAuthority,
+            name: 'My test token',
+            symbol: 'TEST',
+            uri: 'http://test.test',
+        });
+
+        expect(instruction).to.deep.equal(
+            new TransactionInstruction({
+                programId,
+                keys: [
+                    { isSigner: false, isWritable: true, pubkey: metadata },
+                    { isSigner: false, isWritable: false, pubkey: updateAuthority },
+                    { isSigner: false, isWritable: false, pubkey: mint },
+                    { isSigner: true, isWritable: false, pubkey: mintAuthority },
+                ],
+                data: Buffer.from([
+                    // Output of rust implimentation
+                    210, 225, 30, 162, 88, 184, 77, 141, 13, 0, 0, 0, 77, 121, 32, 116, 101, 115, 116, 32, 116, 111,
+                    107, 101, 110, 4, 0, 0, 0, 84, 69, 83, 84, 16, 0, 0, 0, 104, 116, 116, 112, 58, 47, 47, 116, 101,
+                    115, 116, 46, 116, 101, 115, 116,
```

Comment:
You pulled the bytes from the Rust implementation and matched them up here? Impressive! 🙌🏼

Reply:
Yeah - Thought it was the best way to make sure the code was working correct!